batch2/batch/front_end/front_end.py

```diff
@@ -283,7 +285,20 @@
                 if 'memory' not in resources:
                     resources['memory'] = BATCH_JOB_DEFAULT_MEMORY
 
-                cores_mcpu = parse_cpu_in_mcpu(resources['cpu'])
+                req_cores_mcpu = parse_cpu_in_mcpu(resources['cpu'])
+                req_memory_bytes = parse_memory_in_bytes(resources['memory'])
+
+                if req_cores_mcpu == 0:
+                    raise web.HTTPBadRequest(
+                        reason=f'resource requests for job {id} are unsatisfiable: '
+                        f'cpu must be greater than 0')
```

Comment:
0 seems satisfiable, we just don't allow it.

Reply:
Sorry to nit, this still doesn't read well.  Resource requests are always allowed, just that these ones are bad.  How about: "bad resource request for job {id}: cpu cannot be 0".